app/logic/Mail/EWS/EWSAccount.ts

```diff
@@ -203,7 +203,20 @@ export class EWSAccount extends MailAccount {
     let response = await appGlobal.remoteApp.postHTTP(this.url, this.request2XML(aRequest), "text", this.createRequestOptions());
     response.responseXML = this.parseXML(response.data);
     if (response.status == 200) {
-      return this.checkResponse(response, aRequest);
+      try {
+        return this.checkResponse(response, aRequest);
+      } catch (ex) {
+        if (ex.type != "ErrorServerBusy") {
```

Comment:
The Error message says 
`'ConcurrentSyncCalls', policy value '5',`

Can you please:
* Consider the generic `ErrorServerBusy`, and do something sensible, even if we cannot parse the values?
* Please also check for 'ConcurrentSyncCalls' before parsing the values?
* Also parse the value 5. Given the context, I assume them to mean that they accept only 5 parallel calls. That should be something built into `callEWS()`, a throtteling to not have too many  concurrent calls. The default could be something like 10, and then if the server complaints with this error message, we honor this value and do only 5 parallel calls.
* Good idea to also honor the backoff time, as you're doing below. But there should be a sensible upper limit. I've seen backoffs from Office365 that are 1 hour or 1 day. Obviously, if we were to honor that, the app would simply hang for the end user. So, the maximum backup that we honor should be something like 20 seconds.

Reply:
I don't know what you mean by "parsing the values". The backoff time is explicitly provided as a separate XML element; I'm not parsing it from the message.

Reply:
> I don't know what you mean by "parsing the values".

1. Backoff time (as you already do)
2. max number of concurrent calls, in the error message: `policy value '5'`, see my last comment point 3. (we don't check that at all: neither read the value, nor have a throttle of this kind. but we need it.)

Reply:
>     1. Backoff time (as you already do)

No, I don't parse the backoff time.

Reply:
`error` has a number of child fields:
- ResponseClass (always "Error" by the time we get here)
- MessageText (localised error message)
- ResponseCode (always "ErrorServerBusy" by the time we get here)
- DescriptiveLinkKey (not actually used by Exchange; always 0)
- MessageXml (contains additional information that depends on the error)

In the case of an ErrorServerBusy this last field then contains a Value which contains a Value which is the back off time in milliseconds, but it can contain other information e.g. if you try to write to a read-only field then it will contain the field that you were trying to write.

Reply:
The full error message is:
`Error: The server cannot service this request right now. Try again later., This operation exceeds the throttling budget for policy part 'ConcurrentSyncCalls', policy value '5',  Budget type: 'Ews'.  Suggested backoff time 5000 ms`

If the message contains `'ConcurrentSyncCalls'`, can we please pick that `policy value '5'` out of there and parse it into an integer and use it as max concurrentSyncCalls?

Reply:
I understand that this is ugly, but this is Microsoft. It with the microsoft.com domain.